### PR TITLE
Poland can into space

### PR DESCRIPTION
**Pull request by jesings**

### CODE REVIEW

File: src/enemy/enemy_type.rs [hunk 1]
```diff
@@ -36,9 +71,32 @@ impl EnemyType {
                 sequitur.move_pattern_index = (sequitur.move_pattern_index + 1) % CIRCLE.len();
                 rv
             }
+            EnemyType::Polandball(ref mut polska) => {
+                if polska.move_direction == Dir::CRASH {
+                    polska.move_direction = match rand::random::<u8>() % 4 {
+                        0 => Dir::WEST,
```
Comment (tpeters-playoxygen) [inline src/enemy/enemy_type.rs:77]: wait i thought you serialized the enum so you wouldn/t have to do a match statement like this here wasn't that the whole point of assigning them the numerical values

Comment (jesings) [reply src/enemy/enemy_type.rs:77]: A) rong account
B) unfortunately rust lets us map from enums to int discriminants but not back 
C) I don't know what you mean by serialized

File: src/server_gameloop.rs [hunk 2]
```diff
@@ -161,16 +163,24 @@ pub fn gameloop() {
             for enemyid in enemies {
                 let mut enemy = gd.enemies[enemyid].lock().unwrap();
                 let moveloc = enemy.enemy_type.move_pattern();
-                let confirmedloc = enemy.mov(&gd, (Etype::Enemy, enemyid), moveloc, now);
+                assert!(!(*enemy.mut_mov_next() > now));
```
Comment (tpeters-playoxygen) [inline src/server_gameloop.rs]: why tho

**Review by tpeters-playoxygen — APPROVED** [review]
boland ball

File: src/enemy/enemy_type.rs (hunk 1 above)
Comment (tpeters-playoxygen) [reply src/enemy/enemy_type.rs:77]: A) unfortunate
B) unfortunate
C) unfortunate

File: src/server_gameloop.rs (hunk 2 above)
Comment (jesings) [reply src/server_gameloop.rs]: IDK it was mostly to make sure things were sane after I changed it. I can remove if you want

Comment (tpeters-playoxygen) [reply src/server_gameloop.rs]: i would remove, shouldn't be necessary outside of sanity checks while you were implementing the featuers

AND THINK ABOUT THE PERFORMANCE DOWNSIDES

**Pull request merged**